gnark/gnark-jni/gnark-eip-2537.go

```diff
@@ -80,6 +105,28 @@ func eip2537blsG1Add(javaInputBuf, javaOutputBuf, javaErrorBuf *C.char, cInputLe
 	return nonMontgomeryMarshalG1(result, javaOutputBuf, errorBuf)
 }
 
+/*
+eip2537blsG1Mul performs a scalar multiplication on a G1 point.
+
+- Input:
+	- javaInputBuf: Pointer to a buffer containing one G1 point and one scalar
+	- javaOutputBuf: Pointer to a buffer where the resulting G1 point will be written
+	- javaErrorBuf: Pointer to a buffer where error messages will be written if an error occurs
+	- cInputLen: Length of the input buffer in bytes
+	- cOutputLen: Length of the output buffer in bytes
+	- cErrorLen: Length of the error buffer in bytes
+- Returns:
+	- zero is returned if successful, result is written to javaOutputBuf
+	- one is returned if there is an error, error message is written to javaErrorBuf
+- Cryptography:
+	- The field elements that comprise the G1 input point must be checked to be canonical.
+	- The input point must be checked to be on the curve and in the correct subgroup.
+- JNI:
+	- javaInputBuf must be at least (EIP2537PreallocateForG1 + EIP2537PreallocateForScalar) bytes
+	- javaOutputBuf must be at least EIP2537PreallocateForG1 bytes to safely store the result
+
+*/
+
 //export eip2537blsG1Mul
 func eip2537blsG1Mul(javaInputBuf, javaOutputBuf, javaErrorBuf *C.char, cInputLen, cOutputLen, cErrorLen C.int) C.int {
```

Comment:
Another minor JNI safety note: cInputLen does not need to match the number of bytes allocated by javaInputBuf on the java side.

The java side could allocate 1MB and its okay because cInputLen will just take the first cInputLen bytes.